container-search/src/main/java/com/yahoo/search/dispatch/Dispatcher.java

```diff
@@ -222,12 +224,28 @@ private void processResponse(Client.GetDocsumsResponseOrError responseOrError) {
         }
 
         private void fill(List<FastHit> hits, byte[] slimeBytes) {
-            Inspector summaries = new SlimeAdapter(BinaryFormat.decode(slimeBytes).get().field("docsums"));
+            com.yahoo.slime.Inspector root = BinaryFormat.decode(slimeBytes).get();
+            Inspector summaries = new SlimeAdapter(root.field("docsums"));
             if ( ! summaries.valid())
```

Comment:
This will lose the error information if we ever return only errors.
It would be (slightly) more robust to parse errors first and then summaries, and only fail the latter if there are no errors already.